engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -64,6 +66,7 @@ private static class IdExecutionEventListener implements ExecutionEventListener
     private final UUID nextExecutionItem;
     private final Map<UUID, FunctionCallInfo> calls = new HashMap<>();
     private final Timer timer;
+    private final TypeOfNode typeOfNode;
```

Comment:
Using node in an instrument is completely fine. But instead of `ExecutionEventListener` we should rewrite the code to [ExecutionEventNodeFactory](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/instrumentation/ExecutionEventNodeFactory.html) - then the `TypeOfNode` will get _"mounted"_ (and adopted) in the tree of other nodes.